ion/integration_test.go

```diff
@@ -689,8 +680,9 @@ func readCurrentValue(t *testing.T, reader Reader) ionItem {
 	}
 
 	currentType := reader.Type()
-	if reader.IsNull() {
-		ionItem.value = append(ionItem.value, textNulls[currentType])
+	// Excluding SymbolType from setting value to 'null.symbol' because readers can read "'null.symbol'" as text: 'null.symbol'.
```

Comment:
> because readers can read "'null.symbol'" as text: 'null.symbol'

I don't think I understand. If a text reader encounters a `null.symbol`, it has an ion type of `symbol`, `IsNull()` is `true`, and `StringValue()` would return `nil`. 

How would a reader read it as text?

Reply:
This stems from this [test](https://github.com/amzn/ion-tests/blob/master/iontestdata/good/non-equivs/symbols.ion#L25-L26).

According to the symbol specs, if a text reader encounters a single quoted text, it has an ion type of symbol, IsNull() is false, and StringValue() would return the text.

In this case, StringValue() will return `null.symbol` which is the same value appended by the test when text reader encounters unquoted text `null.symbol`

Reply:
> According to the symbol specs, if a text reader encounters a single quoted text, it has an ion type of symbol, IsNull() is false, and StringValue() would return the text.

Agreed. I think my confusion stems from the comment's phrasing:

> because readers **can read** "'null.symbol'" as text: 'null.symbol'.

If a `null.symbol` is wrapped in single quotes (`'null.symbol'`), it *must* be read as text, and never as a null.

> In this case, StringValue() will return "null.symbol" which is the same value appended by the test when text reader encounters unquoted text null.symbol

I would expect `reader.StringValue()` to return:
* `"null.symbol"` when the Ion value is `'null.symbol'` (a symbol)
* An `err` (or `nil`, depending on #152) when the Ion value is `null.symbol` (a null)

Is that what you're describing, or am I missing something?

Reply:
> Is that what you're describing, or am I missing something?

That is correct